src/Middleware/ConcurrencyLimiter/src/ConcurrencyMiddlewareServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.ConcurrencyLimiter;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ConcurrencyMiddlewareServiceCollectionExtensions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+        /// <param name="configure">Set the options used by the queue.
+        /// <returns></returns>
+        public static ConcurrencyLimiterBuilder AddConcurrencyLimiter(this IServiceCollection services, Action<QueuePolicyOptions> configure)
```

Comment:
The `QueuePolicyOptions` is only used by the actual policies right? Given that I think the delegates should be properties on `AddQueuePolicy`/`AddStackPolicy`

Reply:
ok, I will move it policy. Are you ok with the overload with `Action<QueuePolicyOptions, IServiceProvider>` or should I just have `Action<QueuePolicyOptions>`?

Reply:
I think we should have both. Plenty of people just want to set an option without reading from the `IServiceProvider`.